pybamm/expression_tree/broadcasts.py

```diff
@@ -33,72 +32,84 @@ def __init__(
         broadcast_domain,
-        auxiliary_domains=None,
+        broadcast_auxiliary_domains=None,
         broadcast_type="full",
         name=None,
     ):
         # Convert child to scalar if it is a number
         if isinstance(child, numbers.Number):
             child = pybamm.Scalar(child)
+        # Convert domain to list if it's a string
+        if isinstance(broadcast_domain, str):
+            broadcast_domain = [broadcast_domain]
 
         if name is None:
             name = "broadcast"
 
         # perform some basic checks and set attributes
-        domain = self.check_and_set_domain_and_broadcast_type(
-            child, broadcast_domain, broadcast_type
+        domain, auxiliary_domains = self.check_and_set_domains(
+            child, broadcast_type, broadcast_domain, broadcast_auxiliary_domains
         )
         self.broadcast_type = broadcast_type
         self.broadcast_domain = broadcast_domain
-        if auxiliary_domains is None:
-            if child.domain != []:
-                auxiliary_domains = {"secondary": child.domain}
-            else:
-                auxiliary_domains = {}
         super().__init__(name, child, domain, auxiliary_domains)
 
-    def check_and_set_domain_and_broadcast_type(
-        self, child, broadcast_domain, broadcast_type
+    def _unary_simplify(self, simplified_child):
+        """ See :meth:`pybamm.UnaryOperator.simplify()`. """
+        return self._unary_new_copy(simplified_child)
+
+
+class PrimaryBroadcast(Broadcast):
+    """A node in the expression tree representing a primary broadcasting operator.
+    Broadcasts in a `primary` dimension only. That is, makes explicit copies
+
+    Parameters
+    ----------
+    child : :class:`Symbol`
+        child node
+    broadcast_domain : iterable of str
+        Primary domain for broadcast. This will become the domain of the symbol
+    name : str
+        name of the node
+
+    **Extends:** :class:`SpatialOperator`
+    """
+
+    def __init__(self, child, broadcast_domain, name=None):
+        super().__init__(child, broadcast_domain, broadcast_type="primary", name=name)
+
+    def check_and_set_domains(
+        self, child, broadcast_type, broadcast_domain, broadcast_auxiliary_domains
     ):
-        """
-        Set broadcast domain and broadcast type, performing basic checks to make sure
-        it is compatible with the child
-        """
-        # Acceptable broadcast types
-        if broadcast_type not in ["primary", "secondary", "full"]:
-            raise KeyError(
-                """Broadcast type must be either: 'primary', 'secondary', or 'full' and
-                not {}""".format(
-                    broadcast_type
-                )
+        "See :meth:`Broadcast.check_and_set_domains`"
+        # Can only do primary broadcast from current collector to electrode or from
+        # electrode to particle
+        if child.domain == []:
+            pass
+        elif child.domain[0] in [
```

Comment:
can we do a similar check that if the domain is current collector, the broadcast domain must be and electrode or separator?